File: src/main/kotlin/specscript/commands/util/Print.kt
package specscript.commands.util

import specscript.language.AnyHandler
import specscript.language.CommandHandler
import specscript.language.ScriptContext
import specscript.util.toDisplayYaml
import tools.jackson.databind.JsonNode

object Print : CommandHandler("Print", "core/util"), AnyHandler {

    override fun execute(data: JsonNode, context: ScriptContext): JsonNode? {

        println(data.toDisplayYaml())

        return null
    }
}

File: typescript/src/commands/print.ts
import type { CommandHandler } from '../language/command-handler.js'
import type { JsonValue } from '../language/types.js'
import { toDisplayString } from '../language/types.js'
import type { ScriptContext } from '../language/context.js'
import { toDisplayYaml } from '../util/yaml.js'

/**
 * Print: writes a value to stdout.
 */
export const Print: CommandHandler = {
  name: 'Print',

  async execute(data: JsonValue, context: ScriptContext): Promise<JsonValue | undefined> {
    const text = typeof data === 'string' ? data : toDisplayYaml(data)
    const writer = context.session.get('stdout') as ((s: string) => void) | undefined
    if (writer) {
      writer(text)
    } else {
      console.log(text)
    }
    return undefined
  },
}
